Explorer/Assets/DCL/UI/ConnectionStatusPanel/ConnectionStatusPanelController.cs

```diff
@@ -0,0 +1,146 @@
+            IMVCManager mvcManager,
+            ICurrentSceneInfo currentSceneInfo,
+            ECSReloadScene ecsReloadScene,
+            IRoomsStatus roomsStatus,
+            World world,
+            Entity playerEntity
+        ) : base(viewFactory)
+        {
+            this.userInAppInitializationFlow = userInAppInitializationFlow;
+            this.mvcManager = mvcManager;
+            this.currentSceneInfo = currentSceneInfo;
+            this.ecsReloadScene = ecsReloadScene;
+            this.roomsStatus = roomsStatus;
+            this.world = world;
+            this.playerEntity = playerEntity;
+        }
+
+        protected override void OnViewInstantiated()
+        {
+            currentSceneInfo.SceneStatus.OnUpdate += SceneStatusOnUpdate;
+            SceneStatusOnUpdate(currentSceneInfo.SceneStatus.Value);
+            Bind(roomsStatus.ConnectionQualityScene, viewInstance.SceneRoom);
+            Bind(roomsStatus.ConnectionQualityIsland, viewInstance.GlobalRoom);
+        }
+
+        private void SceneStatusOnUpdate(ICurrentSceneInfo.Status? obj)
+        {
+            const float DELAY = 5f;
+
+            async UniTaskVoid ShowButtonAsync()
+            {
+                await UniTask.Delay(TimeSpan.FromSeconds(DELAY));
+                viewInstance.Scene.ShowReloadButton(TryReloadScene);
+            }
+
+            if (obj is not { } status)
+            {
+                viewInstance.Scene.HideStatus();
+                return;
+            }
+
+            viewInstance.Scene.ShowStatus(status);
+
+            if (status is ICurrentSceneInfo.Status.Crashed)
+                ShowButtonAsync().Forget();
+        }
+
+        private void Bind(IReadonlyReactiveProperty<ConnectionQuality> value, IStatusEntry statusEntry)
+        {
+            value.OnUpdate += newValue => UpdateStatusEntry(statusEntry, newValue);
+            UpdateStatusEntry(statusEntry, value.Value);
+        }
+
+        private void UpdateStatusEntry(IStatusEntry statusEntry, ConnectionQuality quality)
+        {
+            if (quality is ConnectionQuality.QualityLost)
+            {
+                ShowErrorAsync().Forget();
+                return;
+            }
+
+            var status = StatusFrom(quality);
+            statusEntry.ShowStatus(status);
+        }
+
+        private async UniTaskVoid ShowErrorAsync()
+        {
+            await mvcManager.ShowAsync(new ShowCommand<ErrorPopupView, ErrorPopupData>(ErrorPopupData.Empty));
+            await userInAppInitializationFlow.ExecuteAsync(true, true, true, world, playerEntity, cancellationTokenSource.Token);
+        }
+
+        protected override UniTask WaitForCloseIntentAsync(CancellationToken ct) =>
+            UniTask.Never(ct);
+
+        public override void Dispose()
+        {
+            currentSceneInfo.SceneStatus.OnUpdate -= SceneStatusOnUpdate;
+            base.Dispose();
+
+            try { cancellationTokenSource.Dispose(); }
```

Comment:
You can use the extension method: `cancellationTokenSource.SafeCancelAndDispose()`